ksql-engine/src/main/java/io/confluent/ksql/function/UdfTemplate.java

```diff
@@ -49,8 +49,10 @@ static String generateCode(final Method method, final String obj) {
         .mapToObj(i -> "arg" + i)
         .collect(Collectors.joining(", "));
 
-    code.addStatement("return (($T) $L).$L($L)",
-                      method.getDeclaringClass(), obj, method.getName(), args);
+    // use the canonical name instead of relying on $T because there is a bug
+    // in javapoet when trying to compile using the default package
```

Comment:
Personally, I'd not bother with the comment.  If there's a test in there that fails if someone reverts this, then I don't think the comment adds any benefit.  I would tag the test with the Java Poet issue link though.